### PR TITLE
EVG-20018: remove unused `Logs` field from the `Task` struct

### PR DESCRIPTION
**Pull request by julianedwards**
EVG-20018

### Description
This is just cleans up the `Task` model by removing the unused `Logs` field.

### Testing
Updated unit tests and checked that task logs are loaded successfully in staging.

### Documentation
N/A

### CODE REVIEW

File: model/task/task.go [hunk 1]
```diff
@@ -82,18 +82,17 @@ type Task struct {
...
 	ContainerAllocatedTime time.Time `bson:"container_allocated_time,omitempty" json:"container_allocated_time,omitempty"`
 
-	Version           string              `bson:"version" json:"version,omitempty"`
-	Project           string              `bson:"branch" json:"branch,omitempty"`
-	Revision          string              `bson:"gitspec" json:"gitspec"`
-	Priority          int64               `bson:"priority" json:"priority"`
-	TaskGroup         string              `bson:"task_group" json:"task_group"`
-	TaskGroupMaxHosts int                 `bson:"task_group_max_hosts,omitempty" json:"task_group_max_hosts,omitempty"`
-	TaskGroupOrder    int                 `bson:"task_group_order,omitempty" json:"task_group_order,omitempty"`
-	Logs              *apimodels.TaskLogs `bson:"logs,omitempty" json:"logs,omitempty"`
```
Comment (ybrill) [inline model/task/task.go:92]: Should we remove [the type](https://github.com/evergreen-ci/evergreen/blob/9fd0768efec624b4c77318fa07b3551c1529f2bb/apimodels/agent_models.go#L77-L81) too?

Comment (ybrill) [reply model/task/task.go:92]: 🤔 I can't find where it's used.
Could you link the specific line?

Comment (ybrill) [reply model/task/task.go:92]: 🤔 I can't find where it's used.
Could you link the specific line?

Comment (julianedwards) [reply model/task/task.go:92]: The evergreen sender is used [here](https://github.com/evergreen-ci/evergreen/blob/main/agent/internal/client/base_client.go#L478). We only use it for local dev, I have [EVG-17662](https://jira.mongodb.org/browse/EVG-17662) to replace it. The `apimodels.TaskLogs` type is used by the agent communicator's [SendLogMessages implementation](https://github.com/evergreen-ci/evergreen/blob/main/agent/internal/client/base_client.go#L492) used, in turn, by the evergreen log sender [here](https://github.com/evergreen-ci/evergreen/blob/main/agent/internal/client/evergreen_sender.go#L76).

Comment (julianedwards) [reply model/task/task.go:92]: LOL we have something called `TaskLogs` and another thing `TaskLog`. My bad.

Comment (julianedwards) [reply model/task/task.go:92]: Done. Good catch!

Comment (ybrill) [reply model/task/task.go:92]: thank you! Sorry to belabor the point.

**Review by ybrill — APPROVED** [review]

**Pull request merged**